ExtraDry/ExtraDry.Core/Models/TaxonomyEntity.cs

```diff
@@ -17,7 +17,7 @@ public void SetParent(T? parent)
                 throw new InvalidOperationException("Parent must be at a higher level than current entity.");
             }
             Ancestors.Clear();
-            Ancestors.AddRange(parent.Ancestors);
+            Ancestors.AddRange(parent.Ancestors.Where(a => a.Uuid != parent.Uuid));
```

Comment:
Wouldn't this be the same as deleting the next line?

Reply:
It should be. I had it separate for explicitness, but you're right, it's not needed. When I deleted the next line, I found that there was actually a bug. SetParent was never being called in the service for items at the top level, so they never had any ancestors. This line was hiding the fact that the "Global" region didn't have a self reference. So I've made a change in the service to make sure that it's called for all items, and simplified here. Everything is looking good again.